material-stepper/src/main/java/com/stepstone/stepper/StepperLayout.java

```diff
@@ -496,6 +509,11 @@ private void onNext() {
         if (verifyCurrentStep(step)) {
             return;
         }
+
+        //if moving forward and got no errors, set hasError to false, so we can have the tab with the check mark.
+        if(mStepperType instanceof TabsStepperType)
```

Comment:
I've given this a lot of thought also, my solution really is not the best, but I was thinking if adding a flag just to use in one place a good approach here... 
Also, it didn't seem to me that in the near future a new kind of tab or something like it would be needed, making a lot of spaghetti 'if's'.

Also it's a no problem to change it !